include/tallies/legendre_polynomial.hpp

```diff
@@ -0,0 +1,31 @@
+#ifndef LEGENDRE_POLYNOMIALS_H
+#define LEGENDRE_POLYNOMIALS_H
+
+#include <vector>
+
+//===================================
+// Legendre Polynomials from order 0 to N order.
+//
+class LegendrePolynomials {
+ public:
+  LegendrePolynomials() = default;
+  LegendrePolynomials(std::size_t order);
+
+  //~LegendrePolynomials() = default;
+
+  std::vector<double> evaluate_legendres(const double x) const;
+
+ private:
+  std::vector<std::vector<double>> legendre_coeff_;
```

Comment:
We should use an `xt::xtensor<double, 2>` for this instead of a vector of vectors. It will be much faster and nicer to our CPU cache. We should also make this change in the ZernikePolynomial if it is also there.

We also might want to consider templating on the order and then making legendre_coeff_ a static variable, that way it is only stored once, but let's make the other changes first.

Reply:
Instead of 2D dimension, I have incorporated in one-dimension and altered the class accordingly. I can confirm that the evaluation of Legendre and Zernike Polynomial is working correctly.